control/navigation.c

```diff
@@ -921,7 +954,7 @@ task_return_t navigation_update(navigation_t* navigation)
 			break;
 			
 		case MAV_STATE_ACTIVE:
-			if((mode_local.byte & 0b11011100) == MAV_MODE_ATTITUDE_CONTROL)
+			if(((mode_local.byte & 0b11011100) == MAV_MODE_ATTITUDE_CONTROL) && (!navigation->throw_recovery_enabled))
```

Comment:
Remember we didn't want the goal coordinates to be updated all the time if we are trying to lock position.